src/controllers/colorjsproxy.cpp

```diff
@@ -24,10 +25,9 @@ QScriptValue ColorJSProxy::colorFromHexCode(uint colorCode) {
     return jsColor;
 }
 
-QScriptValue ColorJSProxy::makeHotcueColorPalette(
-        QScriptEngine* pScriptEngine, UserSettingsPointer pConfig) {
+QScriptValue ColorJSProxy::makeHotcueColorPalette(QScriptEngine* pScriptEngine,
+        HotcueColorPaletteSettings colorPaletteSettings) {
```

Comment:
Make `HotcueColorPaletteSettings::setHotcueColorPalette` emit a signal, make this function a slot, and rename this function to `slotColorPaletteUpdated`.